### ISSUE TITLE
Thorlabs APT Version - Conflict

### ISSUE THREAD

**Issue by dylanjsaunders** [qpit/thorlabs_apt#2]
Hi,

I downloaded the most recent version of thorlabs APT from the website for 64bit windows, using the 64bit version. This is version 3.7.0. However, when doing so, the APT.dll was not present in the directory. However, in the archive section, I could download 3.6.0 and this 64bit version, for windows 64bit did have the APT.dll in the stated directory. 

I am using Python 2.7.9 64bit on Windows 7.

However, now I am getting an error: 

```
Traceback (most recent call last):
  File "<pyshell#3>", line 1, in <module>
    import thorlabs_apt as apt
  File "C:\Users\LocalAdmin\Downloads\thorlabs_apt-master\thorlabs_apt-master\thorlabs_apt\__init__.py", line 1, in <module>
    from .core import *
  File "C:\Users\LocalAdmin\Downloads\thorlabs_apt-master\thorlabs_apt-master\thorlabs_apt\core.py", line 1, in <module>
    from . import _APTAPI
ImportError: cannot import name _APTAPI
```

Any suggestions?

**Comment by tobiasgehring** [comment]
Hi Dylan,

thanks for the hint with the new thorlabs APT version. I'll check it out.

Which python version do you use?

Cheers,

Tobias

**Comment by dylanjsaunders** [comment]
Thanks for getting back so quickly.

2.7

Dylan

**Comment by tobiasgehring** [comment]
Dear Dylan,

it's a compatibility issue between python 2 and python 3. In core.py I do

from . import _APTAPI
from . import _error_codes

which works in python3 but not in python2. What works in python 2 but
not in 3 is

import _APTAPI
import _error_codes

Right now I have no idea how to do this in a way which works with both
python versions. When I have figured it out, I'll update the package. I
hope that until then you could just change the two lines in core.py.
Sorry for the inconvenience.

Thanks for the hint about version 3.7.0. I'll have to look into this.
Maybe thorlabs decided not to support apt.dll any more?

Cheers,

Tobias

**Comment by tobiasgehring** [comment]
Dear Dylan,

could you try to add

from **future** import absolute_import

in the very beginning of core.py (otherwise the original import
statement) and tell me if it works?

I did some tests and the behaviour in python 2 seems to be very erratic.
Sometimes it works and sometimes I get the error you have seen.

Cheers,

Tobias

**Comment by dylanjsaunders** [comment]
Hi Tobias,

Sorry for the delay, especially considering how fast you were to getting to
help me the first time.

I will try adding your suggestion when I am in the lab latter today.
Furthermore, in order to make the code work, I had to add the following to
my python script before importing thorlabs_apt:

import ctypes.util
import thorlabs_apt as apt

If I just imported ctypes it would not work, it would complain that it did
not have the method util.

Also, when I load the thorlabs_apt package, it only seems to work half the
time. I was wondering if there was a command I should add to the end of my
script to ensure that the connection with the device is closed. However, I
could not see one when I looked through core.py. Is there such a command?

Many thanks,
Dylan
